### PR TITLE
[12.0][ADD] hr_expense_payment_difference

### PR DESCRIPTION
**Pull request by Saran440**
cc : @kittiu 
Add function Payment Difference on hr_expense wizard

### CODE REVIEW

File: hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    )
+    payment_difference_handling = fields.Selection([
+        ('open', 'Keep open'),
+        ('reconcile', 'Mark invoice as fully paid')],
+        default='open',
+        string="Payment Difference Handling",
+        copy=False,
+    )
+    writeoff_account_id = fields.Many2one(
+        'account.account',
```
Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]:
```suggestion
        comodel_name='account.account',
```

File: hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        string="Difference Account",
```
Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]:
```suggestion
        string='Difference Account',
```

File: hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        domain=[('deprecated', '=', False)],
+        copy=False,
+    )
+    writeoff_label = fields.Char(
+        string='Journal Item Label',
+        default='Write-Off',
+        help='Change label of the counterpart \
```
Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]:
```suggestion
        help="Change label of the counterpart "
        "that will hold the payment difference",
```

File: hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            HrExpenseSheetRegisterPaymentWizard, self)._get_payment_vals()
+        res.update({
+            'payment_difference_handling': self.payment_difference_handling,
+            'writeoff_account_id': self.writeoff_account_id.id,
+            'writeoff_label': self.writeoff_label,
+        })
+        return res
+
+    @api.multi
+    def _compute_payment_amount(self, invoices=None, currency=None):
```
Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]: Why don't you call super() ?

Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]: I think this is the new method, but inspired by the same method in account.payment, right?
Do you need param - invoices and currency? I don't see you are using it.

File: hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        active_id = self._context.get('active_id', False)
+        expense_sheet = self.env['hr.expense.sheet'].search([
```
Comment (kittiu) [inline hr_expense_payment_difference/models/hr_expense_sheet_register_payment.py]:
```suggestion
        expense_sheet = self.env['hr.expense.sheet'].browse(active_id)
```

File: hr_expense_payment_difference/readme/DESCRIPTION.rst [hunk 6]
```diff
@@ -0,0 +1,2 @@
+This module extends the functionality of hr expense to allow you to create
```
Comment (kittiu) [inline hr_expense_payment_difference/readme/DESCRIPTION.rst]: This module extend the functionality of payment wizard in hr expense, to also allow you to enter the payment difference amount.

**Comment by kittiu** [comment]
Functional test OK.
@Saran440 please edit the code as reviewed.

**Comment by Saran440** [comment]
@kittiu Fixed. Thank you for your review.

**Comment by ps-tubtim** [comment]
@Saran440  LGTM.

File: hr_expense_payment_difference/README.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+Create payment amount not equal total amount on Expense.
+payment difference will show like a invoice.
+
```
Comment (MeetKD) [inline hr_expense_payment_difference/README.rst:43]: Usage statement is bit confusing it would be better if you can simplify it.

**Comment by Saran440** [comment]
@MeetKD Fixed it. Thank you for your reviewed.

**Review by ps-tubtim — APPROVED** [review]

**Review by kittiu — APPROVED** [review]

**Comment by ps-tubtim** [comment]
@Saran440 Could you merge this PR?

**Pull request merged**